File: task3.js
const args = process.argv;
const str = args.slice(2).join(" ");

const res = [];

if(str.match(/[aeiou]/ig))res.push(str.match(/[aeiou]/ig).join(""))
if(str.match(/[bcdfghjklmnpqrstvwxyz]/ig))res.push(str.match(/[bcdfghjklmnpqrstvwxyz]/ig).join(""))
if(str.match(/[\d]/ig))res.push(str.match(/[\d]/ig).join(""))

process.stdout.write(res.join(" "));

File: task3.rb
str = ''

for i in 0 ... ARGV.length
    str = str + ARGV[i]
end

vowels = []
consonants = []
digits = []
    
str.chars.each do |c|
    if (/[aeiouy]/i).match?(c) then vowels.push(c)
    elsif (/[bcdfghjklmnpqrstvwxz]/i).match?(c) then consonants.push(c)
    elsif (/[\d]/).match?(c) then digits.push(c)
    end 
end

puts vowels.join + ' ' + consonants.join + ' ' + digits.join    